docs/custom_hook.md

```diff
@@ -0,0 +1,179 @@
+Custom Hook is a JavaScript function that allows us to reuse some code throughout the app. It is used to validate form/dialog inputs.
+
+Hook is nothing more than Javascript event handling on the events `onCreate`, `onChange`, `onRender`, `onSave`, `onSaveSuccess`, `onSaveFail` and `onEditLoad`.
+
+Hooks can be used inside the services and tabs placed next to the entity tag. 
+
+The configuration and code for adding a hook to an entity are shown below. 
+
+```
+"hook": {
+  "src": "custom_hook",
+  "type": "external"
+}
+```
+The type key needs to be set as external to webpack the custom extensions. UCC  expects scripts marked as external to follow the ESM syntax for exporting and importing modules. Any other value or not specifying type will default to old requireJS (AMD) import syntax. Aditionally, the custom javascript file and the imported modules will not get webpack-ed.
+
+The custom_hook file's relative path to globalConfig.json is `appserver/static/js/build/custom/custom_hook.js`
+
+### Example:
```

Comment:
Please remove colon in the end here and below as well.

Reply:
Done